### PR TITLE
chore(*) update Envoy to 1.20

### LINKED ISSUE #2713: Upgrade to Envoy 1.20.x
### Summary

Envoy 1.19.0 and 1.19.1 are not compatible with Kuma due to https://github.com/envoyproxy/envoy/issues/17921.

I'm hopeful that we may be able to upgrade to 1.19.2, which would involve

* refusing to serve Envoy 1.19.0 and 1.19.1
* temporarily forking go-control-plane to replace the `upstream_resolvers` field type
* detecting the Envoy client version in the DNS configurer and serving different protobufs for Envoy 1.18.4 and 1.19.2

### PR DESCRIPTION
**Pull request by lobkovilya**
### Summary
based on https://github.com/kumahq/kuma/pull/2965

Since Envoy renamed `v3alpha` to `v3`, we have to vendor the old DNS filter proto:
* pkg/xds/envoy/listeners/v3/compatibility/v3alpha/dns_filter.pb.go

### Full changelog

* Update Envoy to 1.20
* Update go-control-plane `github.com/envoyproxy/go-control-plane v0.9.10-0.20211007073556-ab721e93c590`

### Issues resolved

Fix #2713

### Documentation

- [ ] Link to the website [documentation PR](https://github.com/kumahq/kuma-website/pull/XXX)

### Testing

- [X] Unit tests
- [ ] E2E tests
- [ ] Manual testing on Universal
- [X] Manual testing on Kubernetes 

### Backwards compatibility

- [X] Add `backport-to-stable` label if the code is backwards compatible. Otherwise, list breaking changes.

### CODE REVIEW

**Comment by jpeach** [comment]
@lobkovilya Does this need another review pass?

**Comment by lobkovilya** [comment]
@jpeach yes please, it'd great :)

**Review by jpeach — APPROVED** [review]
Quick, merge this before it needs another rebase!

File: .golangci.yml [hunk 1]
```diff
@@ -84,3 +84,6 @@ issues:
     - linters:
         - staticcheck
       text: "SA1019: package sigs.k8s.io/controller-runtime/pkg/client/fake is deprecated" # https://github.com/kumahq/kuma/issues/2460
+    - linters:
+        - staticcheck
+      text: "SA1019: l.ReusePort is deprecated"
```
[the remark this answers is not in the document]
Comment (lobkovilya) [reply .golangci.yml:89]: https://github.com/kumahq/kuma/issues/3326

**Pull request merged**